### ISSUE TITLE
[4.0] Accessibility - Frontend - if "Text to Speech" is activated, the links and some of the buttons don't work

### ISSUE THREAD

**Issue by paternax** [joomla/joomla-cms#30598]
### Steps to reproduce the issue
- Activate the  Plugin: Additional Accessibility Features -> Option: Both
- Activate in the Frontend the Option: Text to Speech
- Click with the mouse on a link or go with the tab to a link and press enter:  they don't work.
  If you are logged in in the frontend and be in the editing mode of an article the "Cancel" and "Save" Buttons don't work, the "Versions" and "Toggle" Buttons work like expected

Perhaps this is an feature and not a bug and you need an assistive tool to navigate

**Comment by adj9** [comment]
When I enable "Text to Speech" in the frontend, I try to press enter and nothing happens, continue reading the menu item. The same behaviour occurs in a blog, for example, in order to read articles and their meta data.

I tried to make an article from the frontend, changed the category and published it without problems.

As for the "Text to Speech" mode, I find it complicated to get out when I want to click on the link. Do I have to disable the voice, is this the only solution?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30598">issues.joomla.org/tracker/joomla-cms/30598</a>.</sub>

**Comment by chmst** [comment]
Confirmed. I am using Firefox and the Console says: 

![screen shot 2020-09-10 at 09 04 58](https://issues.joomla.org/uploads/1/03d721bf2a8b98d4ee13d9d6e801c08a.png)


<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30598">issues.joomla.org/tracker/joomla-cms/30598</a>.</sub>

**Comment by zwiastunsw** [comment]
Excuse me. Who invented this plugin and put it in Joomla? Who needs this plugin? Who decided about it? This does not solve any accessibility problems! It just confuses people's minds.

**Comment by paternax** [comment]
I don't no who wrote this plugin and why it is integrated in Joomla.

I only found during testing some strange behaviours and reported them in the issue tracker. 

The accessibility is one of the main features in Joomla 4, so it should discussed this topic urgently in the working group(s) for Accessibility.

**Comment by brianteeman** [comment]
@zwiastunsw I guess your memory is failing you

**Comment by brianteeman** [comment]
@zwiastunsw you wrote about this plugin
> Despite my doubts, I support this idea.
> You have chosen a very good script (light, no dependencies). If you provide the configuration (disabling / enabling options, changing the icon) it will be a useful plugin.
> Also consider a version for Joomla 3.x.
> Now it works well.

> Let us be clear:
> 
> 1. Such a plugin is useful and may improve the accessibility of the site, provided that it works well. Brian has chosen a very good script that doesn't cause the errors that widgets of this type cause.
> 2. The plugin does not solve the problems of people with disabilities. People with disabilities use specialised tools (screen readers, screen magnifiers, braille embossers… etc.). But it improves usability for the elderly, people with visual impairments, people with cognitive problems.
> 3. E.g. a blind person will use a screen reader and not the option to change text to speech. But this is a very good solution for the elderly, for people who have difficulty concentrating, and also for those who cannot read at the moment.
> 4. Text enlargement is very useful for many seniors and people who see less well. Such people often do not know about the possibilities of the browser and like to use such tools.
> 5. The contrast version will not replace the correct contrasts on the website. But it can be helpful for people who have weaker eyesight or for people who browse the site, for example in a sunny place.
> 6. WCAG does not require the links to be underlined (they must be highlighted). But underlined links can improve accessibility for visually impaired people or for people who have difficulty concentrating.

ref https://github.com/joomla/joomla-cms/pull/24112

**Comment by zwiastunsw** [comment]
Well, as you can see, my memory is already failing me. I remember, that I supported you adding a script to modify contrasts, highlighting links. I did not remember, unfortunately, that there was also this conversion of speech to text. 
As I was writing then, I supported it despite my reservations, which you reminded me.
Now, the internal discussion has concluded that this is an important solution for accessibility.. And it is not so. And hence my reaction. 
**I apologize to you, Brian!** Good intentions turned out to be bad. 

Of course, I have no objections to the usefulness of this solution, as long as it will not be treated as a way to ensure accessibility.

**Comment by brianteeman** [comment]
and as you saw in the original proposal I put it forward at others requests despite my own reservations

**Comment by chmst** [comment]
I did not take notice of this plugin until some weeks ago, shame on me. 
I have made some tests now. I cannot say anything to the text-to-speech feature. 
Decreasing the text-spacing as well as increasing the font size, cause problems if applied several times.
The a11y icon appears on modal windows in a second instance which causes errors

A big cursors and a guiding line are really good and helpful. 

My preferred solution would be: remove the plugin and integrate these two features into Joomla as part of the core.

**Comment by adj9** [comment]
I've done some testing and the individual features on the frontend don't give me any problems (I'm with Chrome v. 85.0.4183.102 (Official Build) - 64 bit - & Mac 10.15.3).

The features I would improve on the plugin are:
- The icon you can move (left and right). Especially in the backend having it fixed would give problems
- An easier deactivation of Text to Speech. When enabled and clicking on a point on the page repeats the text

Also, if npm's accessibility features allow it, with Text to Speech enabled I would have the field alt of the images read. When I click on an image on the frontend or backend nothing happens.

**Comment by chmst** [comment]
#30689 is related
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30598">issues.joomla.org/tracker/joomla-cms/30598</a>.</sub>

**Comment by infograf768** [comment]
See https://github.com/joomla/joomla-cms/issues/30689#issuecomment-695288680

the main problem for Firefox is the position of the icon/parameters relative to the page and not the window when using revert colours.

**Comment by brianteeman** [comment]
:(
